kernel_gateway/services/kernels/handlers.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import notebook.services.kernels.handlers as notebook_handlers
+
+class TokenAuthorizationMixin(object):
+    '''
+    Adds token authentication to tornado.web.RequestHandlers.
+    '''
+    def prepare(self):
+        '''
+        If kg_auth_token is present in the application settings,
+        raises 401 if the request Authorization header does not contain
+        the value "Token <kg_auth_token>".
+        '''
+        server_token = self.settings.get('kg_auth_token')
```

Comment:
Not that you have to, but I tend to decorate a function with `@property` for settings like this.

```
@property
def kg_auth_token():
  return self.settings.get('kg_auth_token')
```

Then I just use it as 

```
server_token = self.kg_auth_token
```

I'd also prefer to not have the `kg_` prefix on the code itself if it already is part of the kernel gateway. That's a nit though, so feel free to ignore.